### PR TITLE
new problem

### PR DESCRIPTION
**Pull request by geniusgordon**
can create new problem
cannot add tag
cannot add testcase

### CODE REVIEW

File: problem/templates/problem/edit.html [hunk 1]
```diff
@@ -1,108 +1,115 @@
...
+  <div style="width:80%; float:left;">
+    <ul class="nav nav-tabs" role="tablist">
+      <li class="active">
+        <a href="#info" role="tab" data-toggle="tab">Info</a>
+      </li>
+      <li>
+        <a href="#description" role="tab" data-toggle="tab">Description</a>
+      </li>
+      <li>
+        <a href="#sampio" role="tab" data-toggle="tab">Sample IO</a>
```
Comment (jim800121) [inline problem/templates/problem/edit.html]: "sampio" may change to "sample_io" will be more readable.

File: problem/templates/problem/edit.html [hunk 2]
```diff
@@ -1,108 +1,115 @@
...
+          <script>
+          CKEDITOR.replace('des')
+          CKEDITOR.replace('inText')
+          CKEDITOR.replace('outText')
+          </script>
+        </div>
+        <div class="tab-pane" id="sampio">
+          <div class='panel panel-default' style='float:left;width:47%'>
+            <div class='panel-heading'>Sample Input</div>
+            <textarea name='sample_input' class='panel-body'
```
Comment (jim800121) [inline problem/templates/problem/edit.html]: "sample_input" and next line "sample_in" may name consistently, it is just a suggestion maybe others have different idea.

p.s. same as "sample_input" below

File: problem/templates/problem/editTab/info.html [hunk 3]
```diff
@@ -22,74 +22,7 @@
...
-    function floatError(cb) {
-      if (cb.checked) {
-        $("#error").show();
-      } else {
-        $("#error").hide();
-      }
-    }
-    </script>
-  </tr>
+{{ form.as_table }}
```
Comment (jim800121) [inline problem/templates/problem/editTab/info.html]: For overall of this file your ' and " usage may be same.
e.g. 
<div class='panel panel-default' style='float:right;width:47%'> 
and
<div class="tab-pane" id="tag">
Their quotation usages are different.

File: problem/templates/problem/edit.html [hunk 4]
```diff
@@ -1,108 +1,115 @@
...
+$('#preview_button').click(function() {
+  MyWindow = window.open('{% url "problem:problem.views.preview" %}',
+    'MyWindow',
+    "toolbar=no,location=no,directories=no,status=no,menubar=no, \
+    scrollbars=yes,resizable=yes,width=600,height=30"
+  );
+  return false;
+});
+</script>
+{% endblock %}
```
Comment (jim800121) [inline problem/templates/problem/edit.html]: For overall of this file your ' and " usage may be same.
e.g. 
<div class='panel panel-default' style='float:right;width:47%'>
and
<div class="tab-pane" id="tag">
Their quotation usages are different.

File: problem/templates/problem/editTab/tag.html [hunk 5]
```diff
@@ -34,12 +42,27 @@
...
+    $("#addTag").submit(function(e) {
+        e.preventDefault();
+        return add_new_tag({{pid}});
+    });
+});
+</script>
+{% else %}
+<strong>Save your problem before editing the tags</strong>
+{% endif %}
+
```
Comment (jim800121) [inline problem/templates/problem/editTab/tag.html]: For overall of this file your ' and " usage may be same.
e.g. 
<div class='panel panel-default' style='float:right;width:47%'>
and
<div class="tab-pane" id="tag">
Their quotation usages are different.

File: problem/templates/problem/editTab/testcase.html [hunk 6]
```diff
@@ -72,15 +56,75 @@
...
+              row.hide();
+            }
+        });
+        return false;
+    });
+});
+</script>
+{% else %}
+<strong>Save your problem before uploading testcases</strong>
+{% endif %}
```
Comment (jim800121) [inline problem/templates/problem/editTab/testcase.html]: For overall of this file your ' and " usage may be same.
e.g. 
\<div class='panel panel-default' style='float:right;width:47%'>
and
\<div class="tab-pane" id="tag">
Their quotation usages are different.

File: problem/templates/problem/problem.html [hunk 7]
```diff
@@ -117,4 +121,3 @@ <h4 class='modal-title'>Limits</h4>
   <br>
 </div>
 </body>
```
Comment (jim800121) [inline problem/templates/problem/problem.html]: For overall of this file your ' and " usage may be same.
e.g. 
\<div class='panel panel-default' style='float:right;width:47%'>
and
\<div class="tab-pane" id="tag">
Their quotation usages are different.

File: problem/views.py [hunk 8]
```diff
@@ -45,25 +56,43 @@ def volume(request):
...
-      'input': 'a, b <= 100000000',
-      'output': 'a+b',
-      'samp_input': '1 2\n4 5',
-      'samp_output': '3\n9\n',
-      'tag': [''],
-      'testcase': [{'num': 1, 'time': 1, 'memory': 32}, {'num': 2, 'time': 3, 'memory': 100}],
-    }
-    return render(request, 'problem/detail.html', p)
+def detail(request, pid):
+    print pid
```
Comment (jim800121) [inline problem/views.py]: Did you forget to delete this line?

**Comment by nickelcheng** [comment]
This PR is ok for me. @CHIAHSINHOU please test this. @jim800121 please check if all comments are fixed.

**Comment by CHIAHSINHOU** [comment]
]
I got the error when I visited the website.

**Comment by henryyang42** [comment]
Module ``general_tools`` was refactored to ``utils`` in the former  PR.

File: problem/templates/problem/editTab/tag.html (hunk 5 above)
Comment (jim800121) [inline problem/templates/problem/editTab/tag.html]: Some quotation usages are not fix.

File: problem/templates/problem/editTab/testcase.html (hunk 6 above)
Comment (jim800121) [inline problem/templates/problem/editTab/testcase.html]: Some quotation usages are not fixed.

**Comment by geniusgordon** [comment]
把general tools 改成utils了

File: problem/views.py (hunk 8 above)
Comment (nickelcheng) [inline problem/views.py]: Please delete useless print.

**Comment by nickelcheng** [comment]
@CHIAHSINHOU please test this again.

**Comment by nickelcheng** [comment]
Please fix un-login bug

**Comment by geniusgordon** [comment]
fixed un-login bug

File: problem/templates/problem/edit.html [hunk 9]
```diff
@@ -1,108 +1,115 @@
...
+      </li>
+      <li>
+        <a href="#testcase" role="tab" data-toggle="tab">TestCase</a>
+      </li>
+    </ul>
+
+    <style>
+    .tab-pane {
+      padding-top: 10px;
+    }
```
Comment (bruce3557) [inline problem/templates/problem/edit.html]: prefer put this in .css file

File: problem/templates/problem/edit.html [hunk 10]
```diff
@@ -1,108 +1,115 @@
...
+<script>
+$("#preview_button").click(function() {
+  MyWindow = window.open('{% url "problem:problem.views.preview" %}',
+    "MyWindow",
+    "toolbar=no,location=no,directories=no,status=no,menubar=no, \
+    scrollbars=yes,resizable=yes,width=600,height=30"
+  );
+  return false;
+});
+</script>
```
Comment (bruce3557) [inline problem/templates/problem/edit.html]: prefer put this in .js file

File: problem/templates/problem/editTab/testcase.html [hunk 11]
```diff
@@ -1,41 +1,14 @@
...
+{% if not is_new %}
+<table id="testcase_table">
+  <style>
+    td {
+      padding: 5px;
+    }
+    input {
+      width: 150px;
+    }
+  </style>
```
Comment (bruce3557) [inline problem/templates/problem/editTab/testcase.html]: ditto

File: problem/templates/problem/editTab/testcase.html [hunk 12]
```diff
@@ -72,15 +56,75 @@
...
+            type: 'GET',
+            url: '/problem/'+{{pid}}+'/testcase/'+tid+'/delete/',
+            success: function(data) {
+              row.hide();
+            }
+        });
+        return false;
+    });
+});
+</script>
```
Comment (bruce3557) [inline problem/templates/problem/editTab/testcase.html]: ditto

File: problem/views.py (hunk 8 above)
Comment (bruce3557) [inline problem/views.py]: If it is need, please use log

**Comment by nickelcheng** [comment]
Did you fix comments? Whenever you fix comments, please leave a message, thx.

**Pull request closed without merging**